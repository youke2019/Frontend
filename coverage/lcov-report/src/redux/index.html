<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/redux
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.05% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/84</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>36/72</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="69.7" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 69%;"></div><div class="cover-empty" style="width:31%;"></div></div></td>
	<td data-value="69.7" class="pct medium">69.7%</td>
	<td data-value="33" class="abs medium">23/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="11" class="abs low">1/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file low" data-value="reducers.js"><a href="reducers.js.html">reducers.js</a></td>
	<td data-value="24.49" class="pic low"><div class="chart"><div class="cover-fill" style="width: 24%;"></div><div class="cover-empty" style="width:76%;"></div></div></td>
	<td data-value="24.49" class="pct low">24.49%</td>
	<td data-value="49" class="abs low">12/49</td>
	<td data-value="36" class="pct low">36%</td>
	<td data-value="25" class="abs low">9/25</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="48" class="abs low">12/48</td>
	</tr>

<tr>
	<td class="file high" data-value="state.js"><a href="state.js.html">state.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Aug 01 2019 01:16:28 GMT+0800 (GMT+08:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
<script src="..\..\block-navigation.js"></script>
</body>
</html>
