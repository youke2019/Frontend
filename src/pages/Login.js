import React, {Component} from 'react';
import {Button, View, Linking,Text} from 'react-native';
import axios from 'axios'

export default class Login extends Component {
  state = {
    user : undefined,
    access_token: undefined
  }

  click() {
    Linking.openURL('https://jaccount.sjtu.edu.cn/oauth2/authorize?client_id=EHuqrWEKvazXzTErwPmCX2m1&response_type=code&redirect_uri=http://10.0.2.2:8080/jaccount/login').catch((err) => console.error("linking error",err));
  }

  componentDidMount() {
    console.log('add')
    Linking.addEventListener('url',this.handleOpenURL);
    Linking.getInitialURL().then((url) => {
      if (url) {
        console.log('get')
        this.handleOpenURL({url});
      }
    })
  }

  componentWillUnmount() {
    console.log('remove')
    Linking.removeEventListener('url', this.handleOpenURL);
  }


  handleOpenURL = ({url}) => {
    var [,access_token] = url.match(/\?access_token=(.*)/)
    console.log(url)

    axios.get('http://10.0.2.2:8080/jaccount/profile?access_token='+access_token).then((response) => {
      this.setState({
        user: response.data,
      })
    }).catch((error) => {
      console.log(error)
    })
  }

  render() {
    const {user, access_token} = this.state;
    return (
      <View>
        <Button
          onPress={this.click}
          title="Login"
        />
        <Text>
        欢迎 {user === undefined? undefined:user.name}
        </Text>
      </View>
    )
  }
}