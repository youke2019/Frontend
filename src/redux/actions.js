export const UPDATE_USER_INFO = 'UPDATEUSERINFO'
export const LOAD_USER_INFO = 'LOADUSERINFO'

export const updateUserInfo = (data) => ({
    type:UPDATE_USER_INFO,
    data
})

export const loadUserInfo = (data) => ({
    type:LOAD_USER_INFO,
    data
})

