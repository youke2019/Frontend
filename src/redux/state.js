const initialState = {
    user_info: null,
    course_list: null,
    login_ready:false,
}

export default initialState