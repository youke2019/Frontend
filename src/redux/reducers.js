import { combineReducers } from 'redux'
import {
    UPDATE_USER_INFO,
    LOAD_USER_INFO,
    CLEAR_USER_INFO,
    SEARCH_COURSES,
} from "./actions.js"
import initialState from "./state";
import axios from 'axios'


function userInfoReducer(state=initialState.user_info, action) {
    switch (action.type) {
        case UPDATE_USER_INFO:
            storage.save({
                key: 'user',
                data: action.data
            })
            state = action.data
            return state
        case LOAD_USER_INFO:
            state = action.data
            return state
        case CLEAR_USER_INFO:
            state = null
            return state
        default:
            return state
    }
}

function courseReducer(state=initialState.course_list, action) {
    switch (action.type) {
        case SEARCH_COURSES:
            state = action.data
            return state
        default:
            return state
    }
}

export const combinedReducer = combineReducers({
    user_info : userInfoReducer,
    course_list: courseReducer,
})