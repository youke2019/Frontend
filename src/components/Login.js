import React, {Component} from 'react';
import {Button, View, Linking,Text} from 'react-native';
import axios from 'axios'
import {updateUserInfo} from "../redux/actions";
import {connect} from "react-redux";

const mapStateToProps = state => {
  return {
    visible : state.user_info == null? true:false
  }
}

const mapDispatchToProps = dispatch => {
  return {
    updateUserInfo: data => {
      dispatch(updateUserInfo(data))
    }
  }
}

class Login extends Component {
  click() {
    Linking.openURL('https://jaccount.sjtu.edu.cn/oauth2/authorize?client_id=EHuqrWEKvazXzTErwPmCX2m1&response_type=code&redirect_uri=http://10.0.2.2:8080/jaccount/login').catch((err) => console.error("linking error",err));
  }

  componentDidMount() {
    Linking.addEventListener('url',this.handleOpenURL);
    Linking.getInitialURL().then((url) => {
      if (url) {
        this.handleOpenURL({url});
      }
    })
  }

  componentWillUnmount() {
    Linking.removeEventListener('url', this.handleOpenURL);
  }


  handleOpenURL = ({url}) => {
    var [,access_token] = url.match(/\?access_token=(.*)/)

    axios.get('http://10.0.2.2:8080/jaccount/profile?access_token='+access_token).then((response) => {
      this.props.updateUserInfo(response.data)
    }).catch((error) => {
      console.log(error)
    })
  }

  render() {
    return (
      <View>
        { this.props.visible ?
            <Button
                onPress={this.click}
                title="Login"
            /> : null
        }
      </View>
    )
  }
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Login)