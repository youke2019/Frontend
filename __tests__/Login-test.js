import React from 'react';

import renderer from 'react-test-renderer';
import {createStore} from "redux";
import {combinedReducer} from "../src/redux/reducers";
import initialState from "../src/redux/state";
import {Provider} from "react-redux";
import Login from '../src/pages/Login'

const store = createStore(combinedReducer,initialState)

jest.mock("../src/utils/LocalStorage.js")

test('renders login page correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Login />
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
});
