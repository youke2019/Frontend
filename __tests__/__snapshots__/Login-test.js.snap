// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login page correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../public/images/bg.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFEEE",
          "fontFamily": "字魂17号-萌趣果冻体",
          "fontSize": 90,
        }
      }
    >
      有课
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "orange",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "width": 210,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "字魂70号-灵悦黑体",
                "fontSize": 25,
                "lineHeight": 25,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            用Jaccount登录
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "height": 100,
            "justifyContent": "center",
            "width": 100,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="yellow"
          hidesWhenStopped={true}
          size="large"
        />
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          加载中
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
